tool/ui.re

```diff
@@ -378,6 +388,46 @@ impl FloatInput {
         })
     }
 }
+impl IntInput {
+    fn onclick(self) {
+        let f = self.onclick;
+        f(self.input);
+    }
+    fn onkey(mut self, key: KeyCode, chr: Option<string>) {
+        let mut input = self.input;
+        if key.to_small() == KEY_BACKSPACE.to_small() {
+            input = self.input.slice(0, -1);
+        } else if key.to_small() == KEY_V.to_small() && (LCTRL_PRESSED || RCTRL_PRESSED) {
+            input = f"{self.input}{Tas::get_clipboard()}";
+        } else if key.to_small() == KEY_D.to_small() && (LCTRL_PRESSED || RCTRL_PRESSED) {
+            input = "";
+        } else {
+            match chr {
+                Option::Some(s) => input = f"{self.input}{s}",
+                Option::None => (),
+            }
+        }
+        self.input = "";
+        let mut i = 0;
+        let mut chr = match input.parse_int() {
+            Result::Ok(chr) => chr,
+            Result::Err(e) => return,
+        };
```

Comment:
I think we should do a similar thing to what we do in the float-input and not accept any non-digit characters.